core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -309,7 +310,7 @@ public void onComplete() {
             responseEncoder.writeReset(req.id(), req.streamId(), Http2Error.INTERNAL_ERROR)
                            .addListener(future -> {
                                try (SafeCloseable ignored = RequestContextUtil.pop()) {
-                                   tryComplete(null);
+                                   fail(ProtocolViolationException.get());
```

Comment:
- What do you think of removing the warning since the exception will be logged to `RequestLog`?
- The message would help to debug.
```suggestion
                                   fail(new ProtocolViolationException(ctx.channel() + "..."));
```

Reply:
I've decided to define an explicit exception to handle https://github.com/line/armeria/pull/4791#discussion_r1154019387 more easily instead 👍 
I think the exception name is self-explanatory, but let me know if you think more information should be included